projects/plugins/jetpack/extensions/blocks/dialogue/components/participants-control.js

```diff
@@ -40,20 +73,224 @@ export function ParticipantsControl( { participants, participantSlug, onSelect }
+	const [ editingMode, setEditingMode ] = useState( participant ? EDIT_MODE_SELECTING : EDIT_MODE_ADDING );
+
+	function onActionHandler( forceFocus ) {
+		switch ( editingMode ) {
+			case EDIT_MODE_ADDING: {
+				return onAdd( value, ! useFocusOutsideIsAvailable || forceFocus );
+			}
+
+			case EDIT_MODE_EDITING: {
+				return onUpdate( {
+					slug: participant.slug,
+					label: getParticipantPlainText( value ), // <- store plain participant value.
+					value,
+				}, ! useFocusOutsideIsAvailable || forceFocus );
+			}
+		}
+
+		setAddAutocomplete( false );
+	}
+
+	/*
+	 * Handle when on focus out.
+	 * Add or Select a participant.
+	 */
+	function onFocusOutsideHandler() {
+		// Clean current participant when content is empty.
+		if ( ! value?.length ) {
+			return setAddAutocomplete( false );
+		}
+
+		onActionHandler();
+	}
+
+	const focusOutsideProps = useFocusOutsideWithFallback( onFocusOutsideHandler );
+
+	/**
+	 * Funcion handler when user types participant value.
+	 * It can add a new participan, or add a new one,
+	 * dependeing on the previous values.
+	 *
+	 * @param {string} newValue - New participant value.
+	 * @returns {null} Null
+	 */
+	function onChangeHandler( newValue ) {
+		// If the new value is empty,
+		// activate autocomplete, and emit onClean(),
+		// to clean the current participant.
+		if ( ! newValue?.length ) {
+			setEditingMode( EDIT_MODE_ADDING );
+			setAddAutocomplete( true );
+			return onClean();
+		}
+
+		// Always update the participant value (block attribute).
+		onParticipantChange( newValue );
+
+		const participantByNewValue = getParticipantByValue( participants, newValue );
+
+		// Set editing mode depending on participant value,
+		// and current conversation participant
+		// tied to this Dialogue block.
+		if ( participant ) {
+			if ( participant.value === newValue ) {
+				setEditingMode( EDIT_MODE_SELECTING );
+			} else {
+				setEditingMode( EDIT_MODE_EDITING );
+			}
+		} else if ( participantByNewValue ) {
+			setEditingMode( EDIT_MODE_SELECTING );
+		} else {
+			setEditingMode( EDIT_MODE_ADDING );
+		}
+	}
+
+	// Keep autocomplete options udated.
+	const autocompleter = useMemo( () => {
+		if ( ! showAutocomplete ) {
+			return [];
+		}
+
+		return [ refreshAutocompleter( participants ) ];
+	}, [ participants, showAutocomplete ] );
+
+	useEffect( () => {
+		setEditingMode( participant ? EDIT_MODE_SELECTING : EDIT_MODE_ADDING );
+		setAddAutocomplete( ! participant );
+	}, [ participant ] );
+
+	return (
+		<div
+			className={ classNames( className, {
+				'is-adding-participant': editingMode === EDIT_MODE_ADDING,
+				'is-editing-participant': editingMode === EDIT_MODE_EDITING,
+				'is-selecting-participant': editingMode === EDIT_MODE_SELECTING,
+			} ) }
+			{ ...focusOutsideProps }
+		>
+			<RichText
+				key={ reRenderingKey }
+				tagName="div"
+				value={ value }
+				formattingControls={ [ 'bold', 'italic', 'text-color' ] }
```

Comment:
Let me update it.

Reply:
> Editing a participant, and then clicking outside the block causes the edit to reset. (This is with WordPress 5.6, but without the Gutenberg plugin active, so I'm guessing it's because useFocusOutside isn't available

Forgot to mention that theoretically, Jetpack sites will support `onFocusOutside` soon. I say `theoretically` because the hook is still experimental :-/ I
I've added a fallback trivial function in case it's not defined in the WordPress package.